src/System.Threading/ref/System.Threading.cs

```diff
@@ -346,12 +377,19 @@ public partial class SynchronizationContext
         public SynchronizationContext() { }
         public static System.Threading.SynchronizationContext Current { get { throw null; } }
         public virtual System.Threading.SynchronizationContext CreateCopy() { throw null; }
+        public bool IsWaitNotificationRequired() { throw null; }
         public virtual void OperationCompleted() { }
         public virtual void OperationStarted() { }
         public virtual void Post(System.Threading.SendOrPostCallback d, object state) { }
         public virtual void Send(System.Threading.SendOrPostCallback d, object state) { }
-        [System.Security.SecurityCriticalAttribute]
         public static void SetSynchronizationContext(System.Threading.SynchronizationContext syncContext) { }
+        protected void SetWaitNotificationRequired() { }
+        [System.CLSCompliantAttribute(false)]
+        [System.Runtime.ConstrainedExecution.PrePrepareMethodAttribute]
```

Comment:
>[System.Runtime.ConstrainedExecution.PrePrepareMethodAttribute] [](start = 8, length = 63)

Just curious, are these needed in the ref?

Reply:
Should they be in coreclr too?

Reply:
It looks like this attribute is only relevant to constrained execution regions, which have no effect in CoreCLR